crates/bevy_scene/src/scene.rs

```diff
@@ -18,3 +25,68 @@ impl Scene {
         Self { world }
     }
 }
+
+impl Scene {
+    /// Clone the scene.
+    ///
+    /// This method will return a `SceneSpawnError` if either a type is not registered in the
+    /// provided `AppTypeRegistry` or doesn't reflect the `Component` trait.
+    pub fn clone(&self, type_registry: &AppTypeRegistry) -> Result<Scene, SceneSpawnError> {
+        let mut new_world = World::new();
+        new_world.insert_resource(type_registry.clone());
+        self.write_to_world(&mut new_world)?;
+        new_world.remove_resource::<AppTypeRegistry>();
+        Ok(Self { world: new_world })
+    }
+
+    /// Write the entities and their corresponding components to the given world.
+    ///
+    /// This method will return a `SceneSpawnError` if either a type is not registered
+    /// or doesn't reflect the `Component` trait.
```

Comment:
Should this be:

```suggestion
    /// This method will return a [`SceneSpawnError`] if either a type is not registered
    /// or doesn't reflect the [`Component`] trait.
```

+ links?